File: AdvancedMap.Objective-C/AdvancedMap/MapOptions.m
//
//  MapOptions.m
//  AdvancedMap.Objective-C
//
//  Created by Aare Undo on 22/09/2017.
//  Copyright © 2017 Nutiteq. All rights reserved.
//

#import "MapOptions.h"

@implementation MapOption

- (id)initWithName:(NSString *)name tag:(NSString*)tag value:(BOOL)value {
    self.name = name;
    self.tag = tag;
    self.value = value;
    
    return self;
}

@end

@implementation MapOptions

+ (NSMutableArray *)getList {
    static NSMutableArray *list = nil;
    
    if (list == nil) {
        
        list = [[NSMutableArray alloc] init];
        
        [list addObject:[[MapOption alloc] initWithName:@"Globe view" tag:@"globe" value:NO]];
        [list addObject:[[MapOption alloc] initWithName:@"3D buildings" tag:@"buildings3d" value:NO]];
        [list addObject:[[MapOption alloc] initWithName:@"3D texts" tag:@"texts3d" value:YES]];
        [list addObject:[[MapOption alloc] initWithName:@"POIs" tag:@"pois" value:NO]];
    }
    
    return list;
}

@end

File: AdvancedMap.Swift/Feature Demo/MapOptions.swift
//
//  MapOptions.swift
//  AdvancedMap.Swift
//
//  Created by carto on 15/05/2019.
//  Copyright © 2019 CARTO. All rights reserved.
//

import Foundation

class MapOptions {
    
    static var list = [MapOption]()
    
    static func initialize() {
        
        var mapOption = MapOption()
        mapOption.name = "Globe view"
        mapOption.tag = "globe"
        mapOption.value = false
        
        list.append(mapOption)
        
        mapOption = MapOption()
        mapOption.name = "3D buildings"
        mapOption.tag = "buildings3d"
        mapOption.value = false
        
        list.append(mapOption)

        mapOption = MapOption()
        mapOption.name = "3D texts"
        mapOption.tag = "texts3d"
        mapOption.value = true
        
        list.append(mapOption)
    }
}

class MapOption : NSObject {
    
    var name: String!
    
    var tag: String!
    
    var value: Bool!
}
